misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/MDCScopeListener.kt

```diff
@@ -1,34 +1,41 @@
-package misk.tracing.backends.datadog
+package com.squareup.cash.tracing.datadog
 
 import datadog.trace.api.CorrelationIdentifier
-import datadog.trace.context.ScopeListener
 import org.slf4j.MDC
 import wisp.logging.getLogger
 
 /**
  * A scope listener that updates the MDC with the trace and span reference anytime a new scope is
  * activated or closed.
  */
-class MDCScopeListener : ScopeListener {
+// Generic exceptions are caught here to avoid crashing the critical path when setting MDC fails.
+@Suppress("TooGenericExceptionCaught")
+internal class MDCScopeListener {
   companion object {
     val log = getLogger<MDCScopeListener>()
     const val MDC_TRACE_ID = "trace_id"
     const val MDC_SPAN_ID = "span_id"
+    const val MDC_DD_TRACE_ID = "dd.trace_id"
+    const val MDC_DD_SPAN_ID = "dd.span_id"
```

Comment:
Some of our internal libraries are also adding the trace and span IDs to these keys as well, so I added them here too. Happy to remove them if preferred not to add them.